lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -0,0 +1,1242 @@
+  IMSI64_TO_STRING(imsi, imsi_str, imsi_len);
+  directoryd_report_location(imsi_str);
+  OAILOG_DEBUG_UE(LOG_AMF_APP, imsi, " Reported UE location to directoryd\n");
+}
+
+//------------------------------------------------------------------------------
+void amf_ue_context_update_coll_keys(
+    amf_ue_context_t* const amf_ue_context_p,
+    ue_m5gmm_context_s* const ue_context_p,
+    const gnb_ngap_id_key_t gnb_ngap_id_key,
+    const amf_ue_ngap_id_t amf_ue_ngap_id, const imsi64_t imsi,
+    const teid_t amf_teid_n11, const guti_m5_t* const guti_p) {
+  hashtable_rc_t h_rc                 = HASH_TABLE_OK;
+  hash_table_ts_t* amf_state_ue_id_ht = get_amf_ue_state();
+  OAILOG_FUNC_IN(LOG_AMF_APP);
+  OAILOG_TRACE(
+      LOG_AMF_APP,
+      "Existing ue context, old_gnb_ue_ngap_id_key %ld "
+      "old_amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT "old_IMSI " IMSI_64_FMT
+      "old_GUTI " GUTI_FMT "\n",
+      ue_context_p->gnb_ngap_id_key, ue_context_p->amf_ue_ngap_id,
+      ue_context_p->amf_context.imsi64,
+      GUTI_ARG_M5G(&ue_context_p->amf_context._guti));
+
+  if ((gnb_ngap_id_key != INVALID_GNB_UE_NGAP_ID_KEY) &&
+      (ue_context_p->gnb_ngap_id_key != gnb_ngap_id_key)) {
+    h_rc = hashtable_uint64_ts_remove(
+        amf_ue_context_p->gnb_ue_ngap_id_ue_context_htbl,
+        (const hash_key_t) ue_context_p->gnb_ngap_id_key);
+    h_rc = hashtable_uint64_ts_insert(
+        amf_ue_context_p->gnb_ue_ngap_id_ue_context_htbl,
+        (const hash_key_t) gnb_ngap_id_key, amf_ue_ngap_id);
+
+    if (HASH_TABLE_OK != h_rc) {
+      OAILOG_ERROR_UE(
+          LOG_AMF_APP, imsi,
+          "Error could not update this ue context %p "
+          "gnb_ue_ngap_ue_id " GNB_UE_NGAP_ID_FMT
+          "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT " %s\n",
+          ue_context_p, ue_context_p->gnb_ue_ngap_id,
+          ue_context_p->amf_ue_ngap_id, hashtable_rc_code2string(h_rc));
+    }
+    ue_context_p->gnb_ngap_id_key = gnb_ngap_id_key;
+  }
+
+  if (amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID) {
+    if (ue_context_p->amf_ue_ngap_id != amf_ue_ngap_id) {
+      h_rc = hashtable_ts_remove(
+          amf_state_ue_id_ht, (const hash_key_t) ue_context_p->amf_ue_ngap_id,
+          (void**) &ue_context_p);
+      h_rc = hashtable_ts_insert(
+          amf_state_ue_id_ht, (const hash_key_t) amf_ue_ngap_id,
+          (void*) ue_context_p);
+
+      if (HASH_TABLE_OK != h_rc) {
+        // TODO: this method is deprecated and will be removed once the AMF's
+        // context is migrated to map in the upcoming multi-UE PR
+        OAILOG_ERROR(
+            LOG_AMF_APP,
+            "Insertion of Hash entry failed for  "
+            "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT PRIX32 " \n",
+            amf_ue_ngap_id);
+      }
+      ue_context_p->amf_ue_ngap_id = amf_ue_ngap_id;
+    }
+  } else {
+    // TODO: this method is deprecated and will be removed once the AMF's
+    // context is migrated to map in the upcoming multi-UE PR
```

Comment:
is this comment still relevant? This PR already supports multi UE, isn't it?

Reply:
yes, AMF supports multi UE

Reply:
comment addressed, removed commented lines